File: backend/app/schemas/xui.py
from datetime import datetime
from uuid import UUID
from pydantic import BaseModel, Field


class XuiConnectionCreate(BaseModel):
    name: str = Field(..., min_length=1, max_length=120)
    host: str = Field(..., min_length=1)
    port: int = Field(default=3306, ge=1, le=65535)
    db_name: str = Field(..., min_length=1)
    db_user: str = Field(..., min_length=1)
    db_pass: str = Field(..., min_length=1)
    is_default: bool = False


class XuiConnectionUpdate(BaseModel):
    name: str | None = None
    host: str | None = None
    port: int | None = Field(default=None, ge=1, le=65535)
    db_name: str | None = None
    db_user: str | None = None
    db_pass: str | None = None
    is_default: bool | None = None


class XuiConnectionOut(BaseModel):
    id: UUID
    name: str
    host: str
    port: int
    db_name: str
    db_user: str
    is_default: bool
    last_test_at: datetime | None
    last_test_ok: bool | None
    last_test_error: str | None
    detected_version: str | None
    created_at: datetime


class XuiTestResult(BaseModel):
    ok: bool
    version: str | None = None
    error: str | None = None
    servers: list[dict] = []
    bouquets: list[dict] = []
    categories: dict = {}

File: src/lib/api/xui.ts
import { api } from "@/lib/api";

export type PanelType = "auto" | "xtream_codes" | "xui_one" | "xui_r22";

export const PANEL_TYPE_LABELS: Record<PanelType, string> = {
  auto: "Autodetectar",
  xtream_codes: "Xtream Codes",
  xui_one: "XUI ONE",
  xui_r22: "XUI ONE r22+",
};

export type XuiConnection = {
  id: string;
  name: string;
  panel_type: PanelType;
  host: string;
  port: number;
  db_name: string;
  db_user: string;
  is_default: boolean;
  last_test_at: string | null;
  last_test_ok: boolean | null;
  last_test_error: string | null;
  detected_version: string | null;
  created_at: string;
};

export type XuiMeta = {
  ok: boolean;
  version: string;
  structure?: { panel_type?: PanelType; versao?: string };
  servers: { id: number; nome: string }[];
  bouquets: { id: number; nome: string }[];
  categories: {
    live: { id: number; nome: string }[];
    movie: { id: number; nome: string }[];
    series: { id: number; nome: string }[];
  };
};

export type XuiTest = {
  ok: boolean;
  version?: string | null;
  error?: string | null;
  servers?: { id: number; nome: string }[];
  bouquets?: { id: number; nome: string }[];
  categories?: XuiMeta["categories"];
};

export type CreatePanelBody = {
  name: string; panel_type: PanelType;
  host: string; port: number;
  db_name: string; db_user: string; db_pass: string;
  is_default?: boolean;
};

export const xuiApi = {
  list: () => api<XuiConnection[]>("/api/v1/xui-connections"),
  create: (body: CreatePanelBody) =>
    api<XuiConnection>("/api/v1/xui-connections", { method: "POST", body }),
  update: (id: string, body: Partial<CreatePanelBody>) =>
    api<XuiConnection>(`/api/v1/xui-connections/${id}`, { method: "PATCH", body }),
  remove: (id: string) => api<void>(`/api/v1/xui-connections/${id}`, { method: "DELETE" }),
  test: (id: string) => api<XuiTest>(`/api/v1/xui-connections/${id}/test`, { method: "POST" }),
  meta: (id: string) => api<XuiMeta>(`/api/v1/xui-connections/${id}/meta`),
};

// alias amigável — código chamando 'panelApi' também funciona
export const panelApi = xuiApi;
